zkevm-circuits/src/bytecode_circuit/circuit.rs

```diff
@@ -1054,30 +1055,28 @@ impl<F: Field> SubCircuit<F> for BytecodeCircuit<F> {
 
     /// Return the minimum number of rows required to prove the block
     fn min_num_rows_block(block: &witness::Block) -> (usize, usize) {
-        if block.bytecode_map.is_some() {
-            // when enable feature "dual_bytecode", get two sets of bytecodes here.
-            let (first_bytecodes, second_bytecodes) = block.get_bytecodes_for_dual_sub_circuits();
-            let minimum_row: usize = max(
-                first_bytecodes
-                    .iter()
-                    .map(|bytecode| bytecode.bytes.len() + 1)
+        #[cfg(feature = "dual_bytecode")]
```

Comment:
one of `bytecode_map` functionality is used to check if configure feature `dual_bytecode`. code like:
if bytecode_map.is_some(){
 // enable dual_bytecode` feature case
}else{
// default : no feature case
}
this can save multiple code `cfg(feature)` statement like used here.
#[cfg(feature = "dual_bytecode")] { 
....
}
#[cfg(not(feature = "dual_bytecode"))]{
.....
}

Reply:
it's better to directly check using `#[cfg(feature = "dual_bytecode")` than checking that bytecode_map is None.

checking using #[cfg(feature = "dual_bytecode") is done at compile time, which ensures that both cases are handled correctly, since we check with and without the dual_bytecode feature enabled. This prevents other people from mistakenly setting `bytecode_map = Some(...)` when the dual_bytecode feature is enabled.

Reply:
at the first, tryied to make all `dual_bytecode` related changes to use #[cfg(feature = "dual_bytecode"), at refactoring stage,  we think many feature cfg checking code is not good even though it is not at compile time, not good for long time maintain purpose.   try to reduce major of them, includking "is_first_bytecode_table" as  default(not feature related,  default value is true), as long as it is compatible with existing functionality.